kernel/packages/scene-system/stateful.scene.system.ts

```diff
@@ -0,0 +1,77 @@
+import { EventSubscriber, WebWorkerTransport } from 'decentraland-rpc'
+import { inject, Script } from 'decentraland-rpc/lib/client/Script'
+import { ILogOpts, ScriptingTransport } from 'decentraland-rpc/lib/common/json-rpc/types'
+import { IEngineAPI } from 'shared/apis/EngineAPI'
+import { ParcelIdentity } from 'shared/apis/ParcelIdentity'
+import { SceneStateStorageController } from 'shared/apis/SceneStateStorageController'
+import { defaultLogger } from 'shared/logger'
+import { DevToolsAdapter } from './sdk/DevToolsAdapter'
+import { RendererActor } from './stateful-scene/RendererActor'
+import { SceneStateDefinition } from './stateful-scene/SceneStateDefinition'
+
+class StatefulWebWorkerScene extends Script {
+  @inject('DevTools')
+  devTools: any
+
+  @inject('EngineAPI')
+  engine!: IEngineAPI
+
+  @inject('ParcelIdentity')
+  parcelIdentity!: ParcelIdentity
+
+  @inject('SceneStateStorageController')
+  sceneStateStorage!: SceneStateStorageController
+
+  private devToolsAdapter!: DevToolsAdapter
+  private renderer!: RendererActor
+  private sceneState!: SceneStateDefinition
+  private eventSubscriber!: EventSubscriber
+
+  constructor(transport: ScriptingTransport, opt?: ILogOpts) {
+    super(transport, opt)
+  }
+
+  async systemDidEnable(): Promise<void> {
+    this.devToolsAdapter = new DevToolsAdapter(this.devTools)
+    const { cid: sceneId } = await this.parcelIdentity.getParcel()
+    this.renderer = new RendererActor(this.engine, sceneId)
+    this.eventSubscriber = new EventSubscriber(this.engine)
+
+    // Fetch stored scene
+    this.sceneState = SceneStateDefinition.fromStorableFormat(await this.sceneStateStorage.getStoredState(sceneId))
+
+    // Listen to the renderer and update the local scene state
+    this.renderer.forwardChangesTo(this.sceneState)
+
+    // Send the initial state ot the renderer
+    this.sceneState.sendStateTo(this.renderer)
```

Comment:
This to me seems like a "piping/stream" idea. We might want to explore the posibility of having an abstraction that "connects" different state containers and makes sure they are synced.